### PR TITLE
[FIX] Update token list when wallet change

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
Token detection not working when fresh install the app and import a fresh account by private key

**Proposed Solution**
Fetch the token list when the address changes

**Test Cases**
Case1:
* Fresh install
* Import wallet with SRP
* wallet with tokens detected
* import wallet via Private key
* Expected tokens be detected as well
* import tokens on both wallet

**Screenshots/Recordings**
IOS: https://recordit.co/yS8vK1Bffh
Android: https://recordit.co/hB5SdcSLs0

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #https://github.com/MetaMask/mobile-planning/issues/614

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: patches/@metamask+assets-controllers+1.0.1.patch [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
++    constructor({ chainId,selectedAddress, preventPollingOnNetworkRestart = false, onNetworkStateChange, onPreferencesStateChange,interval = DEFAULT_INTERVAL, cacheRefreshThreshold = DEFAULT_THRESHOLD, messenger, state, }) {
+         super({
+             name,
+             metadata,
+@@ -78,6 +78,14 @@ class TokenListController extends base_controller_1.BaseControllerV2 {
+                 }
+             }));
+         }
++        if(onPreferencesStateChange){
++        onPreferencesStateChange(({ selectedAddress }) => {
```
Comment (blackdevelopa) [inline patches/@metamask+assets-controllers+1.0.1.patch:19]: Would it be nice if we have this change in the controller (core) repo?

Comment (tommasini) [reply patches/@metamask+assets-controllers+1.0.1.patch]: https://github.com/MetaMask/core/pull/1088 -> Created

**Review by sethkfman — DISMISSED** [review]
LGTM

**Comment by tommasini** [comment]
This PR will be deleted and will be created the same PR pointing to main

**Comment by tommasini** [comment]
I'm closing this PR because it's pointed to 5.14.0 and also the solution presented can be improved, will open a new PR soon wiith a new investigation.
With slow internet conection this PR doesn't solve the problem that is intended to solve
https://recordit.co/CzxQIqFqnF

**Pull request closed without merging**